ast.go

```diff
@@ -67,10 +68,13 @@ func (fa *ASTFunc) generate() {
 	fa.generateReceiver()
 
 	if fa.f.Member != nil {
-		if fa.f.ReturnType.IsSlice {
+		if fa.f.ReturnType.IsSlice || fa.f.ReturnType.IsArray {
```

Comment:
Please always sort such conditions. We do not need micro-optimizations. Clearer code is more important.